lib/commands/reshim.bash

```diff
@@ -88,22 +88,31 @@ write_shim_script() {
   local shim_path
   shim_path="$(asdf_data_dir)/shims/$executable_name"
 
+  local temp_dir
+  temp_dir=${TMPDIR:-/tmp}
+
+  local temp_versions_path
+  temp_versions_path=$(mktemp "$temp_dir/asdf-command-reshim-write-shims.XXXXXX")
```

Comment:
I'm not sure line 91-92 are necessary as `mktemp` should fallback to `TMPDIR` and `/tmp` automatically.

Reply:
@jthegedus, do you have any thoughts since you added this?

Reply:
This seemed fine to me, but after looking at existing use of `mktemp` this was the outlier, so I changed it to align with the other usages. We can back this out later once we clean up and decide to put this behind a function or something.